packages/access-client/src/stores/store-indexeddb.js

```diff
@@ -41,6 +40,10 @@ export class StoreIndexedDB {
     this.#dbStoreName = options.dbStoreName ?? STORE_NAME
   }
 
+  /**
+   *
+   * @returns {Promise<Store>}
```

Comment:
If you declare that these instance methods return a `Store` then you lose the ability to use it in specifics. It should adhere to that interface _anyway_, you shouldn't need to explicitly state it.

There's a bunch of these that have been added - do we really need to?

Reply:
This `Store` type is the local one defined at the top of the file, its already specific. If you dont do this you loose the method documentation defined in the types.ts file for the interface.

What are we losing here ?

Reply:
Oh ok I thought this was referring to the interface!